staging/src/k8s.io/apiserver/pkg/storage/cacher/ready.go

```diff
@@ -89,18 +95,22 @@ func (r *ready) waitAndReadGeneration(ctx context.Context) (int, error) {
 
 		r.lock.RLock()
 		if r.state == Pending {
-			// since we allow to switch between the states Pending and Ready
-			// if there is a quick transition from Pending -> Ready -> Pending
-			// a process that was waiting can get unblocked and see a Pending
-			// state again. If the state is Pending we have to wait again to
-			// avoid an inconsistent state on the system, with some processes not
-			// waiting despite the state moved back to Pending.
 			r.lock.RUnlock()
 			continue
 		}
+
 		generation, err := r.readGenerationLocked()
 		r.lock.RUnlock()
-		return generation, err
+
+		if err != nil {
+			if errors.Is(err, ErrStorageInitializing) {
+				audit.AddAuditAnnotationForRejectWithReason(ctx, "storage_initializing")
+				audit.AddAuditAnnotationForRejectMessage(ctx, err.Error())
```

Comment:
This codepath isn't used when the ResilientWatchCacheInitialization feature is enabled, is it? I think it would be simplest to insert a call to audit.AddAuditAnnotation(...) at https://github.com/kubernetes/kubernetes/blob/a4f14e57f4850541937659dc1cbfc1a5e5efaebf/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L524 and not try to handle any other cases.